test/Streamly/Test/Internal/Data/Parser.hs

```diff
@@ -450,6 +450,41 @@ sliceSepBy =
 --         Right _ -> False
 --         Left _ -> True)
 
+many :: Property
+many = 
+    forAll (listOf (chooseInt (0, 1))) $ \ls ->
+        let
+            concatFold = FL.Fold (\concatList curr_list -> return $ concatList ++ curr_list) (return []) return
+            prsr = P.many concatFold $ P.sliceSepBy (== 1) FL.toList
```

Comment:
Can we have a separate test for `Control.Applicative.many` as well? Also, for `Control.Applicative.some`.